### PR TITLE
implement support for DBIGFONT lump in FON2 format

### LINKED ISSUE #1559: Partial DBIGFONT support
I think it would be a good idea to have support for DBIGFONT in the case of the main settings menu (plus submenu titles) and for custom episodes menus. Would that be something within Woof's scope?

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1559 

Dimension of the Boomed:
woof0000

Eviternity II:
woof0001

### CODE REVIEW

File: src/mn_font.c [hunk 1]
```diff
@@ -0,0 +1,213 @@
+//
+// Copyright(C) 2024 Roman Fomin
```
Comment (fabiangreffrath) [inline src/mn_font.c:2]: Is this all your own code?

Comment (rfomin) [reply src/mn_font.c:2]: I started with the SLADE function, but modified it a lot.

File: src/mn_menu.c [hunk 2]
```diff
@@ -1871,7 +1872,7 @@ void MN_SetNextMenuAlt(ss_types type)
 
 static void M_DrawSetup(void)
 {
-    MN_DrawTitle(124, 15, "M_OPTTTL", "OPTIONS");
+    MN_DrawTitle(124, 15, "M_OPTION", "OPTIONS");
```
[the remark this answers is not in the document]
Comment (rfomin) [reply src/mn_menu.c:1875]: In Eviternity there is no patch for `M_OPTTTL`, so it shows the patch from IWAD. I think `M_OPTION` is more consistent in most cases.

File: src/mn_menu.c [hunk 3]
```diff
@@ -3125,8 +3132,11 @@ void M_Drawer(void)
         {
             if (alttext)
             {
-                MN_DrawStringCR(x, y + 8 - MN_StringHeight(alttext) / 2, cr,
-                                NULL, alttext);
+                if (!MN_DrawFon2String(x, y, cr, alttext))
```
Comment (fabiangreffrath) [inline src/mn_menu.c:3135]: So, if there are patches then we draw patches, else if there is a big font then we use the big font, else we use the hud font?

Comment (rfomin) [reply src/mn_menu.c:3135]: Yes, correct.

File: src/mn_font.c (hunk 1 above)
Comment (rfomin) [reply src/mn_font.c:4]: Is this the right way?https://github.com/fabiangreffrath/woof/pull/1575/commits/57875dd5b6fa5d93b8d4d66dffe560dd45000922

File: src/mn_menu.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ceski-1) [reply src/mn_menu.c:1875]: How about check for `M_OPTTTL` and `M_OPTION`, using the first one that's not an iwad lump?

[the remark this answers is not in the document]
Comment (rfomin) [reply src/mn_menu.c:1875]: > How about check for `M_OPTTTL` and `M_OPTION`, using the first one that's not an iwad lump?

It doesn't help in the DBP37 case, which is why we use `M_OPTTTL` in the first place.

File: src/mn_font.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/mn_font.c:4]: If slade is also licensed under gpl-2, then yes. Could you also add him to the copyright holders in README.md, please?

Comment (rfomin) [reply src/mn_font.c:4]: Done in https://github.com/fabiangreffrath/woof/pull/1575/commits/bd36d3cdc3fcb2831a9f2612b56f5e7ec1062be0

File: src/mn_menu.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply src/mn_menu.c:1875]: Could you leave a comment about this in the code, please?

File: src/mn_font.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/mn_font.c:4]: Thanks, but could you move the copyright line into the `Files: *` block, please? In general, if code is under the same license and doesn't work as a stand-alone module without our own code, then it should go there.

Comment (rfomin) [reply src/mn_font.c:4]: Done in https://github.com/fabiangreffrath/woof/pull/1575/commits/154423e82d12cd20221b09a991102352dc87adff

File: src/mn_menu.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (rfomin) [reply src/mn_menu.c]: On second thought, I don't know of any PWAD with custom menu patches that don't have M_OPTTTL, except Eviternity 1/2. I revert this in  https://github.com/fabiangreffrath/woof/pull/1575/commits/fa936dc8bb7432ce8ecf55f09073354aaabc0e98

**Comment by fabiangreffrath** [comment]
> adapt V_LinearToTransPatch from Eternity Engine
> This will be useful for PNG support.

This rings my ear!

**Comment by fabiangreffrath** [comment]
Some WADs have sky textures in PNG format, so this would be useful at last.

**Comment by fabiangreffrath** [comment]
I remember a map that crashed as soon as the sky was rendered, because that was composed of a PNG patch. I think we draw it as plain black now, but with this code you added, we could at least draw it as a paletted approximation.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**